reo/process_results.py

```diff
@@ -70,6 +74,280 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
+                macrs_basis = macrs_bonus_basis * (1 - wind['macrs_bonus_pct'])
+                depreciation_schedule = np.array([0.0 for _ in range(years)])
+                for i,r in enumerate(schedule):
+                    depreciation_schedule[i] = macrs_basis * r
+                depreciation_schedule[0] += (wind['macrs_bonus_pct'] * macrs_bonus_basis)
+                total_depreciation += depreciation_schedule
+
+        #calculate Storage capital costs, o+m costs, incentives, and depreciation
+        if (storage['size_kw'] or 0) > 0:
+            total_kw = storage.get('size_kw') or 0
+            total_kwh = storage.get('size_kwh') or 0
+            capital_costs = (total_kw * storage['installed_cost_us_dollars_per_kw']) + (total_kwh * storage['installed_cost_us_dollars_per_kwh'])
+            battery_replacement_year = int(storage['battery_replacement_year'])
+            battery_replacement_cost = -1 * ((total_kw * storage['replace_cost_us_dollars_per_kw']) + (total_kwh * storage['replace_cost_us_dollars_per_kwh']))
+            om_series += np.array([0 if yr != battery_replacement_year else battery_replacement_cost for yr in range(1, years+1)])
+            
+            #storage only has cbi in the API
+            cbi = (total_kw * storage['total_rebate_us_dollars_per_kw']) + (total_kwh * (storage.get('total_rebate_us_dollars_per_kw') or 0)) 
+            total_ibi_and_cbi += (cbi)
+            # Depreciation
+            if storage['macrs_option_years'] in [5,7]:
+                if storage['macrs_option_years'] == 5:
+                    schedule = macrs_five_year
+                if storage['macrs_option_years'] == 7:
+                    schedule = macrs_seven_year
+                federal_itc_basis = capital_costs - cbi
+                federal_itc_amount = storage['total_itc_pct']*federal_itc_basis
+                federal_itc += federal_itc_amount
+                macrs_bonus_basis = federal_itc_basis - (federal_itc_basis * storage['total_itc_pct'] * storage['macrs_itc_reduction'])
+                macrs_basis = macrs_bonus_basis * (1 - storage['macrs_bonus_pct'])
+                depreciation_schedule = np.array([0.0 for _ in range(years)])
+                for i,r in enumerate(schedule):
+                    depreciation_schedule[i] = macrs_basis * r
+                depreciation_schedule[0] += (storage['macrs_bonus_pct'] * macrs_bonus_basis)
+                total_depreciation += depreciation_schedule
+
+        #calculate Generator capital costs, o+m costs, incentives, and depreciation
+        if (generator['size_kw'] or 0) > 0:
+            total_kw = generator.get('size_kw') or 0
+            new_kw = (generator.get('size_kw') or 0) - (generator.get('existing_kw') or 0)
+            existing_kw = generator.get('existing_kw') or 0
+            
+            capital_costs = (new_kw * generator['installed_cost_us_dollars_per_kw'])
```

Comment:
the generator capital_costs are never used. where are any of the tech capital costs accounted for in the cash flow?

Reply:
@NLaws we get capital costs from initial_capital_costs in the Financial outputs